dist/function.py

```diff
@@ -0,0 +1,131 @@
+# sort before non-pointers.
+def function_args_alpha(text):
+        s = text.strip()
+        s = re.sub("[*]","", s)
+        s = re.sub("^const ","", s)
+        s = re.sub("^volatile ","", s)
+        return s
+
+# Return the sort order of a variable declaration, or no-match.
+#       This order isn't defensible: it's roughly how WiredTiger looked when we
+# settled on a style, and it's roughly what the KNF/BSD styles look like.
+def function_args(line):
+    types = {
+        'u_int16_t ' : -1,      # Not allowed
+        'u_int32_t ' : -1,      # Not allowed
+        'u_int64_t ' : -1,      # Not allowed
+        'u_int8_t ' : -1,       # Not allowed
+        'u_quad ' : -1,         # Not allowed
+        'uint ' : -1,           # Not allowed
+        'struct ' : 1,          # Types in argument display order. */
+        'union ' : 2,
+        'enum ' : 3,
+        'WT_' : 4,
+        'wt_' : 5,
+        'double ' : 6,
+        'float ' : 7,
+        'size_t ' : 8,
+        'uint64_t ' : 9,
+        'int64_t ' : 10,
+        'uint32_t ' : 11,
+        'int32_t ' : 12,
+        'uint8_t ' : 13,
+        'int8_t ' : 14,
+        'u_int ' : 15,
+        'int ' : 16,
+        'u_char ' : 17,
+        'char ' : 18,
+        'bool ' : 19,
+        'va_list ' : 20,
+        'void ' : 21,
+
+    }
+    line = line.strip()
+    line = re.sub("^const ", "", line)
+    line = re.sub("^volatile ", "", line)
+
+    # Let WT_UNUSED terminate the parse. It often appears at the beginning
+    # of the function and looks like a WT_XXX variable declaration.
+    if re.search('^WT_UNUSED', line):
+        return False,0
+
+    # Let lines not terminated with a semicolon terminate the parse, if means
+    # there's some kind of interesting line split we probably can't handle.
+    if not re.search(';$', line):
+        return False,0
+
+    # Check for matching types.
+    for m in types:
+        # Don't list '{' as a legal character in a declaration, that's what
+        # prevents us from sorting inline union/struct declarations.
+        if re.search('^' + m + "\s*[\w(*]", line):
+            return True,types[m]
+    return False,0
+
+# Put function arguments in correct sort order.
+def function_declaration():
+    tmp_file = '__tmp'
+    for name in all_c_files():
+        skip_re = re.compile(r'DO NOT EDIT: automatically built')
+        s = open(name, 'r').read()
+        if skip_re.search(s):
+            continue
+
+        # Read through the file, and for each function, do a style pass over
+        # the local declarations. Quit tracking declarations as soon as we
+        # find anything we don't understand, leaving it untouched.
+        with open(name, 'r') as f:
+            tfile = open(tmp_file, 'w')
+            tracking = False
+            for line in f:
+                if not tracking:
+                    tfile.write(line)
+                    if re.search('^{$', line):
+                        r = [[] for i in range(30)]
```

Comment:
I am wondering if we can get rid of "30" here. How about having a variable containing the count of illegal types, say illegal_type_count = 6, declared with "types". Then instead of "30" we can have "len(types) - illegal_type_count". That said, I am fine with how it is at the moment too, it is very readable.

Reply:
> I am wondering if we can get rid of "30" here. How about having a variable containing the count of illegal types, say illegal_type_count = 6, declared with "types". Then instead of "30" we can have "len(types) - illegal_type_count". That said, I am fine with how it is at the moment too, it is very readable.

Agreed, fixed in 4b7f599.

I ended up using enumerate and getting rid of the specific indexes, but my Python is weak, this one probably deserves a second look.